lib/fli/FliObjHdl.cpp

```diff
@@ -0,0 +1,530 @@
+{
+    if (m_const) {
+        LOG_ERROR("Attempted to set a constant signal/variable!\n");
+        return -1;
+    }
+
+    if (m_num_elems == 1) {
+        mtiInt32T enumVal = m_enum_map[value.c_str()[0]];
+
+        if (m_is_var) {
+            mti_SetVarValue(get_handle<mtiVariableIdT>(), enumVal);
+        } else {
+            mti_SetSignalValue(get_handle<mtiSignalIdT>(), enumVal);
+        }
+    } else {
+        int len = value.length();
+        int numPad;
+
+        numPad = len < m_num_elems ? m_num_elems-len : 0;
+
+        LOG_DEBUG("set_signal_value(string)::%s", value.c_str());
+
+        if (len > m_num_elems) {
+            LOG_DEBUG("FLI: Attempt to write sting longer than (%s) signal %d > %d", m_name.c_str(), len, m_num_elems);
+            len = m_num_elems;
+        }
+
+        char *iter = (char *)m_mti_buff;
+        mtiInt32T enumVal;
+        std::string::iterator valIter;
+        int i = 0;
+
+        // Pad MSB for descending vector
+        for (i = 0; i < numPad && !m_ascending; i++) {
+            iter[i] = (char)m_enum_map['0']; 
+        }
+
+        for (valIter = value.begin(); (valIter != value.end()) && (i < m_num_elems); valIter++, i++) {
+            enumVal = m_enum_map[*valIter];
+            iter[i] = (char)enumVal; 
+        }
+
+        // Fill bits a the end of the value to 0's
+        for (i = len; i < m_num_elems && m_ascending; i++) {
+            iter[i] = (char)m_enum_map['0']; 
+        }
+
+        if (m_is_var) {
+            mti_SetVarValue(get_handle<mtiVariableIdT>(), (long)iter);
+        } else {
+            mti_SetSignalValue(get_handle<mtiSignalIdT>(), (long)iter);
+        }
+    }
+
+    return 0;
+}
+
+int FliIntObjHdl::initialise(std::string &name, std::string &fq_name)
+{
+    m_num_elems   = 1;
+
+    m_val_buff = (char*)malloc(33);  // Integers are always 32-bits
+    if (!m_val_buff) {
+        LOG_CRITICAL("Unable to alloc mem for value object read buffer: ABORTING");
+        return -1;
+    }
+    m_val_buff[m_num_elems] = '\0';
+ 
+    return FliValueObjHdl::initialise(name, fq_name);
+}
+
+const char* FliIntObjHdl::get_signal_value_binstr(void)
+{
+    mtiInt32T val;
+
+    if (m_is_var) {
+        val = mti_GetVarValue(get_handle<mtiVariableIdT>());
+    } else {
+        val = mti_GetSignalValue(get_handle<mtiSignalIdT>());
+    }
+
+    std::bitset<32> value((unsigned long)val);
+    std::string bin_str = value.to_string<char,std::string::traits_type, std::string::allocator_type>();
+    snprintf(m_val_buff, 33, "%s", bin_str.c_str());
+
+    return m_val_buff;
+}
+
+long FliIntObjHdl::get_signal_value_long(void)
+{
+    mtiInt32T value;
+
+    if (m_is_var) {
+        value = mti_GetVarValue(get_handle<mtiVariableIdT>());
+    } else {
+        value = mti_GetSignalValue(get_handle<mtiSignalIdT>());
+    }
+
+    return (long)value;
+}
+
+int FliIntObjHdl::set_signal_value(const long value)
+{
+    if (m_const) {
```

Comment:
Python does not allow a _setcachedvalue on a ConstantObject so these checks may be redundant.